frontend/src/metabase/lib/form.ts

```diff
@@ -0,0 +1,24 @@
+import { t } from "ttag";
+import _ from "lodash";
+import { BaseFieldDefinition } from "metabase-types/forms";
+
+export function createErrorMessageMarkdown(
+  message?: string,
+  formFields?: BaseFieldDefinition[],
+): string | undefined {
+  if (!message) {
+    return undefined;
+  }
+
+  const fieldTitles = (formFields || [])
+    .map(formField => formField.title)
+    .filter(_.identity) as string[];
+
+  return fieldTitles.reduce((transformedMessage, fieldTitle) => {
```

Comment:
Still return a markdown because it makes the util testable. I'm not sure how to test if the util produces the desired React node though.